columnflow/inference/cms/datacard.py

```diff
@@ -352,15 +351,19 @@ def write_shapes(
         # create the output file
         out_file = uproot.recreate(shapes_path)
 
-        # helper to fill empty bins in-place
-        def fill_empty(h):
-            value = h.view().value
-            mask = value <= 0
-            value[mask] = fill_empty_bins
-            h.view().variance[mask] = fill_empty_bins
-
         # iterate through shapes
         for cat_name, hists in self.histograms.items():
+            cat_obj = self.inference_model_inst.get_category(cat_name)
+
+            # helper to fill empty bins in-place
+            fill_empty = lambda h: None
+            if cat_obj.fill_empty_bins:
+                def fill_empty(h):
+                    value = h.view().value
+                    mask = value <= 0
+                    value[mask] = cat_obj.fill_empty_bins
+                    h.view().variance[mask] = cat_obj.fill_empty_bins
```

Comment:
See other comment about `fill_empty_bins`, might be better to call it `fill_empty_bins_with` or `default_fill_value` or something similar

Reply:
Yup, renamed :+1: